docs/getting_started/how_to_build_an_api.md

```diff
@@ -0,0 +1,215 @@
+
+1. open a terminal, navigate to `examples/getting_started/api`, and type `$ go run main.go`
+2. open a second terminal, and type `curl localhost:8080/euro/10/GBP` or `curl localhost:8080/euro-at-date/10/USD/2021-01-03T00:00:00Z`
+
+To understand what _money-exchange_ does, let us start by looking at its _main_ function:
+
+[embedmd]:# (../../examples/getting_started/api/main.go /func main/ /\n}/)
+```go
+func main() {
+	app := application.New(
+		application.WithConfigFile("config.dist.yml", "yml"),
+		application.WithKernelSettingsFromConfig,
+		application.WithLoggerHandlersFromConfig)
+
+	app.Add("api", apiserver.New(definer.ApiDefiner))
+	app.Add("currency", currency.NewCurrencyModule())
+
+	app.Run()
+}
+```
+
+This application creates a kernel, using _config.dist.yml_ for its configuration, adds module _"currency"_ and _"api_ to it, then starts the kernel.
+
+These are the contents of _config.dist.yml_:
+```
+env: dev
+
+app_project: gosoline
+app_family: example
+app_name: money-exchange
+
+kernel:
+  killTimeout: 1s
+
+kvstore:
+  currency:
+    type: chain
+    in_memory:
+      max_size: 500000
+    application: money-exchange
+    elements: [inMemory]
+    ttl: 30m
+```
+
+The most interesting config is the currency key value store. This is defined as `inMemory` and serves as a local database.
+
+The _"currency"_ module is already defined by Gosoline, and will use this kvstore to store the exchange rates for various currencies. Its main functionalities are:
+- it makes an initial call to an external endpoint to get exchange rates, and stores them in a kvstore
+- it occasionally makes more calls to obtain exchange rates, and keep the kvstore updates
+
+The _"app"_ module is our api server, notice that it was created with a call to `apiserver.New`:
+
+[embedmd]:# (../../examples/getting_started/api/main.go /app.Add\(\"api\"/ /\)/)
+```go
+app.Add("api", apiserver.New(definer.ApiDefiner)
+```
+
+The code for _ApiDefiner_ is as follows:
+
+[embedmd]:# (../../examples/getting_started/api/definer/definer.go /func ApiDefiner/ /\n}/)
+```go
+func ApiDefiner(ctx context.Context, config cfg.Config, logger log.Logger) (*apiserver.Definitions, error) {
+	definitions := &apiserver.Definitions{}
+
+	currencyService, err := currency.New(ctx, config, logger)
+	if err != nil {
+		return nil, fmt.Errorf("can not create currencyService: %w", err)
+	}
+
+	definitions.GET("/euro/:amount/:currency", apiserver.CreateHandler(handler.NewEuroHandler(logger, currencyService)))
+	definitions.GET("/euro-at-date/:amount/:currency/:date", apiserver.CreateHandler(handler.NewEuroAtDateHandler(logger, currencyService)))
+
+	return definitions, nil
+}
+```
+
+_ApiDefiner_ does four things:
+
+1. Creates a _currencyService_ which offers currency exchange functionality by using the kvstore.
+2. It creates a new EuroHandler.
+3. Creates a new EuroAtHandler.
+4. It instantiates a new _api.Definitions_ objects to which it adds to routes served by the two handlers.
```

Comment:
This would be the interesting point I guess. I would expect that you need to talk about:
- that you need to create the definitions object yourself
- what the definitions.GET call does, that there is also .POST, etc.
- what does `:amount` in the path mean? Ah, we capture something, how can we later access that?